api_compatibility_policy.md

```diff
@@ -150,10 +152,51 @@ which includes the earliest date each feature will be removed.
 
 ## Go libraries compatibility policy
 
-Tekton Pipelines may introduce breaking changes to its Go client libraries, as long as these changes
-do not impact users' yaml/json CRD definitions. For example, a change that renames a field of a CRD
-would need to follow the policy on [backwards incompatible changes](#backwards-incompatible-changes),
-but a change that renames the Go struct type for that field is allowed.
+### Covered parts of the codebase
+
+The compatibility policy for the [github.com/tektoncd/pipeline Go package](https://pkg.go.dev/github.com/tektoncd/pipeline) currently only covers
+Go structs representing CRDs in `pkg/apis`, the generated client libraries in `pkg/client`, and the [resolver interface](https://github.com/tektoncd/pipeline/blob/main/pkg/resolution/resolver/framework/interface.go).
+Backwards compatibility for other parts of the codebase is on a best-effort basis.
+
+Pipelines contributors are working on modularizing `pkg/apis`, so that clients only need to import a minimal set of dependencies.
+This work is tracked in [issue #6483](https://github.com/tektoncd/pipeline/issues/6483).
+When it is complete, the compatibility policy will be updated to include all of `pkg/apis`.
+
+Pipelines contributors also plan to move all internal functionality in the entire [github.com/tektoncd/pipeline Go package](https://pkg.go.dev/github.com/tektoncd/pipeline)
+into internal packages. This work is tracked in [issue #5679](https://github.com/tektoncd/pipeline/issues/5679).
+When it is complete, the compatibility policy will be updated to include all of `github.com/tektoncd/pipeline`.
+
+### Alpha CRDs
+
+- Go structs representing alpha CRDs may change in breaking ways at any time. If the change impacts the yaml/JSON API, we will provide one release of warning.
+
+- If support for an alpha CRD is dropped, its client library will not be removed until the last release with support for the CRD has reached its end of life.
+
+- If support for a field is dropped, the field may also be removed from the client libraries; i.e. it is not guaranteed to be tombstoned.
```

Comment:
👍

Reply:
What does it mean to "tombstone" a field? I can guess but I'm not entirely sure and I couldn't easily find a definition online. Perhaps it makes sense to link to a definition?

Reply:
I added a definition here, lmk if it is clear!